src/Hazelcast.Net/CP/CpSubsystemSession.cs

```diff
@@ -0,0 +1,271 @@
+﻿// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Hazelcast.Clustering;
+using Hazelcast.Core;
+using Hazelcast.Exceptions;
+using Microsoft.Extensions.Logging;
+
+namespace Hazelcast.CP
+{
+    /// <summary>
+    /// Cp Subsystem Session manages server side session requests and heartbeat
+    /// </summary>
+    internal partial class CPSubsystemSession : IAsyncDisposable
```

Comment:
We don't use proxy in naming. Also, we don't have much proxy, I think. Session is a kind of service actually. WDYT @zpqrtbnk ?

Reply:
I am slightly confused by the name. A "session" is just, well, a session, not something that manages sessions. Now I respectfully hate that every single class in Java is prefixed `Client` as in `ClientProxyWhatever` because well, yes, we're writing a client, and I am even tempted to retaliate by naming my classes `DotNetClientProxyWhatever`. So please can we drop the `Client` prefix and use a name that makes sense? Like, we are managing sessions, can it be `SessionManager`? In the cp subsystem namespace? or `CpSessionManager` or something that *tells* what it does? And yes it's not going to be the same name as Java but the Java names in most cases make no sense.

Reply:
Renamed to `CpSessionManager` , _cpSubsystemSession to `_cpSessionManager`. Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/590/commits/5172b1429ee358627e0192663ae666a29610b29f